docs/reference/ingress/kubernetes-with-cert-manager.md

```diff
@@ -0,0 +1,168 @@
+# Deploying with SSL on Kubernetes using Cert-Manager and Nginx
+
+The quickest way to get automated SSL/TLS certificates for your project is to use [cert-manager][cert-manager]. In this tutorial, we will deploy OpenFaaS using the [Helm Chart][openfaas-helm], [cert-manager][cert-manager], and [nginx-ingress][nginx-ingress]
+
+**RBAC** We will assume that your cluster is configured for [RBAC][k8s-rbac].
+
+## Configure Helm and Tiller
+
+First install Helm and the Tiller [following the instructions provided by Helm][helm-install]
+
+## Create namespaces
+
+```sh
+$ kubectl create namespace openfaas
+$ kubectl create namespace openfaas-fn
+```
+
+## Install nginx-ingress
+
+Add the `nginx-ingress` using
+
+```sh
+$ helm install stable/nginx-ingress --name nginxingress --set rbac.create=true
+```
+
+The full configuration options for nginx [can be found here][nginx-configuration].
+
+## Install cert-manager
+
+We will treat `cert-manager` like a core component of the cluster, so we install it into the `kube-system` namespace using the following command:
+
+```sh
+$ helm install \
+    --name cert-manager \
+    --namespace kube-system \
+    stable/cert-manager
+```
+
+This configuration will work for most deployments, but you can also see https://cert-manager.readthedocs.io/en/latest/getting-started/2-installing.html for additional instructions and options for installing cert-manager.
+
+## Configure cert-manager
+
+In additional to the controller installed in the previous step, we must also configure an "Issuer" before `cert-manager` can create certificates for our services. For convenience we will create an Issuer for both Let's Encrypt's production API and their staging API. The staging API has much kinder rate limits and is useful for testing.
+
+Replace `<your-email-here>` with the contact email that will be shown with the SSL certificate.
+
+```yaml
+# letsencrypt-issuer.yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-prod
+  namespace: openfaas
+spec:
+  acme:
+    # Email address used for ACME registration
+    email: <your-email-here>
+    http01: {}
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      key: ""
+      name: letsencrypt-prod
+    server: https://acme-v01.api.letsencrypt.org/directory
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+  namespace: openfaas
+spec:
+  acme:
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: <your-email-here>
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    http01: {}
+```
+
+```sh
+$ kubectl apply -f letsencrypt-issuer.yaml
+```
+
+This will allow `cert-manager` to automatically provision Certificates just in the `openfaas` namespace.
+
+## Install openfaas
+
```

Comment:
I think it might be flawed to assume that OpenFaaS isn't already installed. It might be inconvenient to land here and then have to remove everything and start over.

Reply:
Should I change the title here? The helm command provided will upgrade an existing install with the same name _or_ install it.  So "Configure" could be valid here instead.

I think the language of this section becomes much more complicated if you assume that OpenFaaS was previously installed via Helm.  I would have to look something like 

> if you installed with a custom values, `<previous-values>.yaml` then you can upgrade using 
> ```
> helm upgrade openfaas \
>     --install \
>     --namespace openfaas \
>     --values <previous-values>.yaml \
>     --values tls.yaml \
>     openfaas/openfaas
> ```
> 
> If you used `--set` in your previous installation of OpenFaaS, you must ....  If you ....

etc  which seems messy.  If OpenFaaS was not installed with Helm, then I am not sure what to add here, I guess instructions with a plain manifest file?  But I don't think we want to promote using manifest files instead of Helm.   Overall, it becomes much messier.

Reply:
The Ingress definition is pretty simple, we could provide it as a sample in addition?

All I have to add is something like this generally.. 

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: openfaas-ingress
  namespace: openfaas
  annotations:
    kubernetes.io/tls-acme: "true"
    kubernetes.io/ingress.class: "nginx"
    certmanager.k8s.io/cluster-issuer: "letsencrypt-prod"
  labels:
    app: faas-netesd
spec:
  tls:
  - hosts:
    - gateway.domain.io
    secretName: o6s-com-tls
  rules:
  - host: gateway.domain.io
    http:
      paths:
      - path: /
        backend:
          serviceName: gateway
          servicePort: 8080
```